extension/options/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+
+const defaultOptions = {
+  arrowParens: "avoid",
+  bracketSpacing: true,
+  jsxBracketSameLine: false,
+  printWidth: 80,
+  semi: true,
+  singleQuote: false,
+  tabWidth: 2,
+  trailingComma: "none",
+  useTabs: false
+};
+
+function App() {
+  const [options, setOptions] = useState();
+
+  useEffect(() => {
+    window.chrome.storage.sync.get(defaultOptions, setOptions);
+  }, [setOptions]);
+
+  useEffect(() => {
+    if (options) {
+      window.chrome.storage.sync.set(options);
+    }
+  }, [options]);
+
+  function handleChange({ target: { checked, name, type, value } }) {
+    setOptions({
+      ...options,
+      [name]:
+        type === "checkbox"
+          ? checked
+          : type === "number"
+          ? parseInt(value)
+          : value
+    });
+  }
+
+  return options ? (
+    <>
+      <h1>
+        <a href="https://prettier.io/docs/en/options.html">Prettier Options</a>
```

Comment:
Might be good to turn the [`jsx-no-target-blank`]( https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-target-blank.md) rule on.

Reply:
Is there a rule that bans `target="_blank"` completely (not just when it's insecure)? Screenreaders don't consistently implement support for this, so the extra tab/page can cause accessibility issues.

Reply:
Is there a rule that bans `target="_blank"` completely (not just when it's insecure)? Screenreaders don't consistently implement support for this, so the extra tab/page can cause accessibility issues.

Reply:
Whoops, I thought this had the `target="_blank"` attribute. In this case, might it make sense to add it since this will be any options pop up in Chrome?

Reply:
I'm not sure how links are handled in popups, so I'm trying to test it out. I have this in the manifest (instead of `options_page`) and it's not letting me open the popup, any ideas? ```js
  "options_ui": {
    "page": "options/build/index.html",
    "open_in_tab": false
  },
```

Reply:
Hmmm I'm not sure - I haven't actually made a Chrome Extension settings page before. The popup isn't showing up at all?

Reply:
Nope, the button is clicked and nothing shows up, not even an error

Reply:
I still can't get this to work. I'm giving up for now but willing to use a popup if someone else figures it out.